Cookbook/Proposals/ManagingAssets.md

```diff
@@ -0,0 +1,89 @@
+
+## Alternatives considered
+
+1. We could try to systematize the way we include assets in specific feature frameworks but it can cause problems described in the motivation section.
+	
+2. Instead of accessing icons or images by subscripts `designLibrary.tokens.icons[.close]` we could use new feature of Swift 5.1 `@dynamicMemberLookup` which could be combined with `KeyPath`. Then we could write just `designLibrary.tokens.icons.close`.
+To achieve that firstly we have to mark `struct Icons` with `@dynamicMemberLookup`. Then `ImageIdentifier` has to become `struct` with `String` properties with default value:
+```
+struct ImageIdentifier {
+    let close = "close"
+}
+```
+Later we need create property `let imageIdentifier = ImageIdentifier()` inside `struct Icons`. Having all of these we can finally write dynamicMember subscript:
+```
+subscript(dynamicMember keyPath: KeyPath<ImageIdentifier, String>) -> UIImage {
+    return Icons.image(for: imageIdentifier[keyPath: keyPath])
+}
+```
+And finally use it like: `designLibrary.tokens.icons.close`. The only drawback of this approach is the fact that for new icons we have to create new property and assign the default value to it, compared to just create new `enum` `case` with the name matching asset name. The small benefit here is to have a little bit better syntax while keeping adding new assets simple.
+
+3. We are going to use SwiftGen tool to auto-generate localizable strings identifiers (https://github.com/Babylonpartners/ios-playbook/pull/187). Having that tool in place we could also use it to generate assets identifiers.
+
+`SwiftGen` is generating 0-case enum called `Assets` with `static let` corresponding to a given asset:
+```
+enum Asset {
+    static let close = ImageAsset(name: "Close")
+}
+```
+Then you can use it like this:
+```
+let closeImage = UIImage(asset: Asset.close) 
+```
+or 
+```
+let sameCloseImage = Asset.close.image
+```
+Ultimately we would like to limit SwiftGen impact on our codebase to absolute minimum. That's why I think we can use it alongside with `@dynamicMemberLookup`. There are couple of issues which has to be resolved first. If we want to use `KeyPath` we would have to be able to create an instance of `Asset`. It's not possible with 0-case enum that's why we should change it to `struct`. Another thing is to change `static let`s to just `let`s because `Dynamic key path member lookup cannot refer to static member 'close'`. To match this requirements we can just create custom templates in SwiftGen.
```

Comment:
> Ultimately we would like to limit SwiftGen impact on our codebase to absolute minimum.

Why? We already approved its usage for localisable strings and SwiftGen generates an enumeration so the impact is not that significant as we can remove it and keep the same enumeration or not?

Reply:
If we are using a flat structure of single-level enum this is minimal impact and I think that's the same thing we think about. What I meant is to not use nested levels of namespacing which potentially could spread out through codebase. Maybe sometime in the future, Apple will add a tool for this kind of purpose then the transition to it will be simple.